go/pkg/libproxy/multiplexed.go

```diff
@@ -138,10 +138,14 @@ func (c *channel) Write(p []byte) (int, error) {
 			return written, io.EOF
 		}
 		if c.write.size() > 0 {
+			// Some window space is available
 			toWrite := c.write.size()
 			if toWrite > len(p) {
 				toWrite = len(p)
 			}
+			// Advance the window before dropping the lock in case another Write() appears and sees the same available space
+			c.write.current = c.write.current + uint64(toWrite)
```

Comment:
in case of an error, `c.write.current` will be updated as if the data was sent.
This behavior is different from the previous
You can use a temporary variable instead.

Reply:
Unfortunately if there's an error, the whole multiplexer will need to shut down because the messages on the wire will be corrupted. For example if I write:

- Message 1 header: "my payload length is 100 bytes"
- Message 1 payload: only 50 bytes because of an error
- Message 2 header: some other packet

then the remote will think "Message 2 header" is part of "Message 1 payload" and everything breaks :(

So I think in the case of error we need to close everything and report the error upwards.

Reply:
fair enough